### DISCUSSION TITLE
Is it possible to get an Auth user from his email?

### DISCUSSION THREAD

**Discussion by SandroMaglione** [supabase/supabase#19598] [Questions]
I have some code that calls [`auth.signUp`](https://supabase.com/docs/reference/javascript/auth-signup). Sometimes it happens that another operation after it fails and the sign up is not registered in any table inside `public`.

When this happens it becomes impossible from code to retrieve a user from his email from Auth, get his id, and create a row in the database.

Is it possible to check **if a user with a given email exists inside Auth and get his id**?

**Comment by GaryAustin1** [comment]
You can certainly use an rpc call to a postgres function (security definer type) and select from auth.users where the email matches.  The function could return an id or true if user exists.    This also means that anyone can call that function if your intent is to use it from a browser.  Also not really sure what you would do with the id on a browser as the jwt is needed if you have RLS checking for user ids.   If on a server then you could use service_role and the id to get data.

**Reply by SandroMaglione** [reply]
I have a webhook that calls `signUp` to register the user, then gets the `id` of the `User` returned by `signUp` and registers a new user inside a `public` table with `email` and `id`.

If the user is already registered inside Auth **but** without a corresponding row inside a `public` table, then `signUp` fails and no row is added to the table.

Afterwards there is no way to get back the `id` of the user from the email to add back the row inside the table.

What would you suggest here?

**Reply by GaryAustin1** [reply]
How does it come to be that your webhook does not update the public table?  If there is an error you would want to retry.
Otherwise you could certainly check if the email exists with the rpc/function in your webhook before you do the signUp. 
Are you confirming the signUp with email?  If not then it should return an error that user already exists.

Can you use a trigger function like the user management guides show to populate your public table?  That is the normal way to have a user/profile table in the public schema?  https://supabase.com/docs/guides/auth/managing-user-data#using-triggers
